src/content.ts

```diff
@@ -0,0 +1,72 @@
+const fetchData = () => {
+  const emoteData = [];
+  fetch('https://api.betterttv.net/3/cached/emotes/global', {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+    .then((response) => response.json())
+    .then((json) => {
+      json.forEach((x) => {
+        const tempObj = { name: x.code, id: x.id };
+        emoteData.push(tempObj);
+      });
+    });
+
+  return emoteData;
+};
+
+const processMessage = (text, HTML) => {
+  emotes.forEach((x) => {
```

Comment:
```suggestion
  emotes.forEach((emote) => {
```
whenever I use `forEach`, `map`, etc. I try to call the parameter the singular version of what I'm looping through. so `emotes` => `emote`, etc. which is quite common cause typically we do call our arrays some plural of what it contains.

Reply:
You can also do object desctructuring here so instead of calling it `x` or `emote` you can call it `{name, id}` and then you can access `name` and `id` in the code w/out saying `x.name` etc